engine/src/test/java/org/hibernate/search/test/backend/DeleteByQueryTest.java

```diff
@@ -175,24 +205,24 @@ private void testSerializationForQuery(DeletionQuery deletionQuery) {
 	private List<Work> makeBooksForSingularTermQuery() {
 		List<Work> list = new LinkedList<>();
 		// just some random data:
-		list.add( new Work( new Book( String.valueOf( 5 ), "Lord of The Rings" ), WorkType.ADD ) );
-		list.add( new Work( new Book( String.valueOf( 6 ), "The Hobbit" ), WorkType.ADD ) );
+		list.add( new Work( new Book( 1, "Lord of The Rings", "lordoftherings" ), WorkType.ADD ) );
+		list.add( new Work( new Book( 2, "The Hobbit", "thehobbit" ), WorkType.ADD ) );
 		return list;
 	}
 
 	private List<Work> makeBooksForNumRangeQuery() {
 		List<Work> list = new LinkedList<>();
 		// just some random data:
-		list.add( new Work( new Book( String.valueOf( 5 ), 1, 1L, 1F, 1D, "Lord of The Rings" ), WorkType.ADD ) );
-		list.add( new Work( new Book( String.valueOf( 6 ), 2, 2L, 2F, 2D, "The Hobbit" ), WorkType.ADD ) );
+		list.add( new Work( new Book( 1, 1, 1L, 1F, 1D, "Lord of The Rings", "lordoftherings" ), WorkType.ADD ) );
+		list.add( new Work( new Book( 2, 2, 2L, 2F, 2D, "The Hobbit", "thehobbit" ), WorkType.ADD ) );
 		return list;
 	}
 
 	@Indexed(index = "books")
 	private static class Book {
 
 		@DocumentId
-		String id;
+		Integer id;
```

Comment:
Do you we have a test that checks deletion when the id is a String?
I'm a bit surprise that you changed an existing test instead of adding a new one.